runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/StreamingImpulseSource.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.flink.translation.wrappers.streaming.io;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.apache.beam.sdk.util.WindowedValue;
+import org.apache.flink.streaming.api.functions.source.RichParallelSourceFunction;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A streaming source that periodically produces an empty byte array. This is mostly useful for
+ * debugging, or for triggering periodic behavior in a portable pipeline.
+ */
+public class StreamingImpulseSource extends RichParallelSourceFunction<WindowedValue<byte[]>> {
+  private static final Logger LOG = LoggerFactory.getLogger(StreamingImpulseSource.class);
+
+  private final AtomicBoolean cancelled = new AtomicBoolean(false);
+  private long count = 0;
+  private final int intervalMillis;
+  private final int messageCount;
+
+  public StreamingImpulseSource(int intervalMillis, int messageCount) {
+    this.intervalMillis = intervalMillis;
+    this.messageCount = messageCount;
+  }
+
+  @Override
+  public void run(SourceContext<WindowedValue<byte[]>> ctx) {
+    // in order to produce messageCount messages across all parallel subtasks, we divide by
+    // the total number of subtasks
+    int subtaskCount = messageCount / getRuntimeContext().getNumberOfParallelSubtasks();
```

Comment:
It would be good if we can run this "forever", without upper bound. I think that should be the default when the user does not specify a count.

Reply:
Yes, that's the default. If messageCount is 0 we'll run forever.